url65a69497a4_sweider_BankSystem_JavaLab8_tgz-pJ8-banksystem_lab4_LauncherJ8/src/banksystem/lab4/core/summaryChecker/SummaryChecker.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package banksystem.lab4.core.summaryChecker;
+
+import banksystem.lab4.core.bank.IBank;
+import banksystem.lab4.core.broker.IBroker;
+import banksystem.lab4.core.moneyamount.MoneyAmount;
+import java.util.Collection;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author alex
+ */
+public class SummaryChecker implements Runnable{
+    private static final Logger logger = LogManager.getLogger(SummaryChecker.class);
+    private final Collection<Integer> accountList;
+    private final IBroker accountBroker;
+    private final MoneyAmount summary;
+    
+    public SummaryChecker(IBank bank, IBroker broker){
+        this.accountBroker=broker;
+        this.accountList=bank.getIdList();
+        this.summary=bank.getSummary();
+    }
+
+    @Override
+    public void run() {
+        MoneyAmount summ;
+        while(true){
+            summ = new MoneyAmount(0);            
+            this.accountBroker.uncheckAll();
+            
+            for (Integer currenAccount : this.accountList){
+                summ=summ.add(this.accountBroker.getAmountAndSetChecked(currenAccount.intValue()));
+            }
+            
+            if (summ.compareTo(this.summary)==0){
+                logger.info("Summary is valid.");
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)